File: coupon/src/main/java/com/sparta/coupon/application/exception/CouponException.java
package com.sparta.coupon.application.exception;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public class CouponException extends RuntimeException {

    private final Error error;
    private final HttpStatus httpStatus;
}

File: order/src/main/kotlin/com/study/order/application/exception/CouponException.kt
package com.study.order.application.exception

open class CouponException(val error: Error) : RuntimeException()

class InvalidCouponCategoryException : CouponException(Error.INVALID_COUPON_CATEGORY)

class InvalidCouponPriceException : CouponException(Error.INVALID_COUPON_PRICE)

class InvalidCouponException : CouponException(Error.INVALID_COUPON)